### PR TITLE
Account creation

### PR DESCRIPTION
**Pull request by rbosseIII**

### CODE REVIEW

File: src/main/java/com/micetr0/model/Composition.java [hunk 1]
```diff
@@ -18,12 +18,24 @@ public Composition() {
         //empty constructor
     }
 
+    private Integer compositionID;
```
Comment (rbosseIII) [inline src/main/java/com/micetr0/model/Composition.java]: will probably cause merge conflicts. Consider removing.

File: src/main/java/com/micetr0/mock_DB/Mock_DB.java [hunk 2]
```diff
@@ -162,10 +143,18 @@ public void deleteComposition(Integer compositionId)
    @Override
     public void deleteAccount(String username)
    {
-       List<Integer> accId = findAccountIdByUsername(username);
+       List<Account> accIds = findAccountByUsername(username);
+
+       if(accIds.size() < 1){
+           //too many accounts in returned account list, should only be one, throw exception
```
Comment (PatHock) [inline src/main/java/com/micetr0/mock_DB/Mock_DB.java:187]: where is the exception?

Comment (PatHock) [reply src/main/java/com/micetr0/mock_DB/Mock_DB.java:187]: also, should be accIds.size() > 1

File: src/main/java/com/micetr0/mock_DB/Mock_DB.java [hunk 3]
```diff
@@ -162,10 +143,18 @@ public void deleteComposition(Integer compositionId)
    @Override
     public void deleteAccount(String username)
```
Comment (PatHock) [inline src/main/java/com/micetr0/mock_DB/Mock_DB.java:182]: should return Boolean

File: src/main/java/com/micetr0/mock_DB/Mock_DB.java [hunk 4]
```diff
@@ -162,10 +143,18 @@ public void deleteComposition(Integer compositionId)
...
    {
-       List<Integer> accId = findAccountIdByUsername(username);
+       List<Account> accIds = findAccountByUsername(username);
+
+       if(accIds.size() < 1){
+           //too many accounts in returned account list, should only be one, throw exception
+       }
+
+       Account tempAccount = new Account();
+       tempAccount = accIds.get(0);
```
Comment (PatHock) [inline src/main/java/com/micetr0/mock_DB/Mock_DB.java:191]: get rid of tempAccount, not needed

File: src/main/java/com/micetr0/mock_DB/Mock_DB.java [hunk 5]
```diff
@@ -187,15 +180,14 @@ public void insertAccount(Account account)
      * @return List of Account ID's that have the specified username
      */
     @Override
-    public List<Integer> findAccountIdByUsername(String username) {
-        List<Integer> accountIdList = new ArrayList<>();
+    public List<Account> findAccountByUsername(String username) {
+        List<Account> accountIdList = new ArrayList<>();
```
Comment (PatHock) [inline src/main/java/com/micetr0/mock_DB/Mock_DB.java:222]: rename to accountList

File: src/main/java/com/micetr0/mock_DB/MySqlDB.java [hunk 6]
```diff
@@ -258,7 +258,7 @@ public Boolean execute(Connection conn) throws SQLException {
 
 
     @Override
-    public void insertNote(Note note) {
+    public void insertNote(String type, String pitch, Integer measureIndex, Integer measureId) {
```
Comment (PatHock) [inline src/main/java/com/micetr0/mock_DB/MySqlDB.java:262]: should be Boolean

File: src/main/java/com/micetr0/mock_DB/MySqlDB.java [hunk 7]
```diff
@@ -270,19 +270,87 @@ public void deleteNote(String noteId) {
...
+                Boolean found = false;
+
+                while(accResultSet.next()){
+                    found = true;
+
+                    Account account = new Account();
+                    loadAccount(account, accResultSet, 1);
+
+                    resultAccounts.add(account);
+                }
```
Comment (PatHock) [inline src/main/java/com/micetr0/mock_DB/MySqlDB.java:341]: Catch exception here

File: src/main/java/com/micetr0/mock_DB/MySqlDB.java [hunk 8]
```diff
@@ -270,19 +270,87 @@ public void deleteNote(String noteId) {
...
+
+                if(!found){
+                    System.out.println("No Account was found that matched that username");
+                }
+
+                return resultAccounts;
+            }
+            finally{
+                DBUtil.closeQuietly(getAccStmt);
+                DBUtil.closeQuietly(getAccStmt);
```
Comment (PatHock) [inline src/main/java/com/micetr0/mock_DB/MySqlDB.java]: duplicate statement

File: src/main/java/com/micetr0/mock_DB/MySqlDB.java [hunk 9]
```diff
@@ -306,12 +374,68 @@ public void deleteAccount(String username) {
...
+
+                    resultAccounts.add(account);
+                }
+
+                if(!found){
+                    System.out.println("Account was not added successfully");
+                }
+
+                return accountID;
+            }
```
Comment (PatHock) [inline src/main/java/com/micetr0/mock_DB/MySqlDB.java:427]: catch exception(s)

File: src/test/java/com/micetr0/mock_DB/Mock_DBtest.java [hunk 10]
```diff
@@ -42,28 +44,23 @@ void insertNoteTest() {
...
-    {
-        List<Account> Account = db.findCurrentAccount(1);
-        assertEquals("sad_Keanu", Account.get(0).getUsername());
-    }
+    void findAccountByUsernameTest(){
 
+    }
+*/
     @Test
     void findAllAccountsTest()
```
Comment (PatHock) [inline src/test/java/com/micetr0/mock_DB/Mock_DBtest.java:65]: implement.

File: src/test/java/com/micetr0/mock_DB/Mock_DBtest.java [hunk 11]
```diff
@@ -119,19 +117,6 @@ void findAccountIdByUsernameAndPasswordTest() {
 
     }
 
-
-    @Test
-    void findAccountIdByUsernameTest() {
-        // Test sad_Keanu
```
Comment (PatHock) [inline src/test/java/com/micetr0/mock_DB/Mock_DBtest.java]: Why delete a unit test?

**Pull request closed without merging**